sdk/core/scheduler/plic.h

```diff
@@ -139,6 +145,10 @@ namespace sched
 			{
 				if (ConfiguredInterrupts[i].number == uint32_t(source))
 				{
+					if (ConfiguredInterrupts[i].isEdgeTriggered)
```

Comment:
Is this really the right place to do this?  `futex_word_for_source` is called by `interrupt_futex_get`, which is (AIUI) meant to be used in initialization, not necessarily when an interrupt has happened.  Perhaps this belongs in the other caller of `futex_word_for_source`, namely `do_external_interrupt`?

Separately, because we're in IRQ context, and so running with IRQs disabled, I don't think there's concern about the order in which completing the interrupt and incrementing the futex word happens, but it'd be good to document that fact (assuming it is, in fact, a fact).